tests/many_to_one/tests.py

```diff
@@ -667,3 +667,28 @@ def test_add_then_remove_after_prefetch(self):
         self.assertEqual(city.districts.count(), 2)
         city.districts.remove(d2)
         self.assertEqual(city.districts.count(), 1)
+
+    def test_cannot_reuse_trimmed_joins(self):
+        """
+        When a trimmable join is specified in the query (here city__), the ORM
```

Comment:
` tests/many_to_one` looks like it's testing the behavior of managers more than queries. As much as I don't like to make `tests/queries` larger, that might be a better place. I'd describe the purpose of the test more like `test_inverted_q_across_relations` rather than describing it in terms of the underlying implementation details which might change over time.